cmd/swctl/app/cmd_dependency.go

```diff
@@ -0,0 +1,494 @@
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+
+			return nil
+		},
+	}
+	var glob = GlobalOptions{
+		Debug:              false,
+		LogLevel:           "",
+		Color:              "",
+		ComposeFiles:       nil,
+		EntityFiles:        nil,
+		EmbeddedEntityByte: nil,
+	}
+	cli.Initialize(&glob)
+
+	cmd.AddCommand(installExternalTools(cli), dependencyStatus(cli), installHugePages(cli), linkSetUpDown(cli), startupConf(cli))
+
+	return cmd
+}
+
+func dependencyStatus(cli Cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "status",
+		Short:         "status",
+		Args:          cobra.ArbitraryArgs,
+		SilenceErrors: true,
+		SilenceUsage:  true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			docker, err := IsDockerAvailable(cli)
+			if err != nil {
+				return err
+			}
+			hugePages, err := AllocatedHugePages(cli)
+			if err != nil {
+				return err
+
+			}
+			physicalInterfaces, err := DumpDevices(cli)
+			if err != nil {
+				return err
+			}
+			var status string
+			if docker {
+				status = "OK"
+			} else {
+				status = "Not installed"
+			}
+			fmt.Fprintf(cli.Out(), "Docker: %s\n", status)
+
+			if hugePages == 0 {
+				status = "Disabled"
+			} else {
+				status = strconv.Itoa(hugePages)
+			}
+			fmt.Fprintf(cli.Out(), "Hugepages: %s\n", status)
+
+			if physicalInterfaces == nil {
+				status = "No available interfaces\n"
+				fmt.Fprintf(cli.Out(), status)
+			} else {
+				table := tablewriter.NewWriter(cli.Out())
+				table.SetHeader([]string{"Name", "Pci", "Mode", "Driver"})
+
+				for _, n := range physicalInterfaces {
+					row := []string{n.Name, n.Pci}
+					if n.Driver == n.Module {
+						row = append(row, "Kernel")
+					} else {
+						row = append(row, "DPDK")
+					}
+					row = append(row, n.Driver)
+
+					table.Append(row)
+				}
+				table.Render()
+			}
+			return nil
+		},
+	}
+	return cmd
+}
+
+func installExternalTools(cli Cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install-tools",
+		Short: "Install external tools",
+		Args:  cobra.ArbitraryArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			docker, err := IsDockerAvailable(cli)
+			if err != nil {
+				return err
+			}
+
+			if !docker {
+				err = InstallDocker(cli, "default")
+				if err != nil {
+					return err
+				}
+			}
+			fmt.Println("Docker is already installed")
+
+			return nil
+		},
+	}
+	return cmd
+}
+
+func installHugePages(cli Cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "set-hugepages",
+		Short: "set-hugepages <value>",
+		Args:  cobra.ArbitraryArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			size, err := strconv.Atoi(args[0])
+			if err != nil {
+				return err
+			}
+			if err = ResizeHugePages(cli, uint(size)); err != nil {
+				return err
+			}
+			return nil
+
+		},
+	}
+	return cmd
+}
+
+/* DPDK interface cannot be used by kernel otherwise it won't connect to VPP*/
```

Comment:
I missed this before. This will take golang as function documentation, but this is definitely not documentation for this function (it should start with function name and state its purpose like "linkSetUpDown changes the link state and binds/unbinds the pci driver (up=kernel driver usable for kernel network stack, down=no kernel driver). The kernel driver unbinding is helpfull in case of DPDK interfaces that can't have kernel network stack usable driver when VPP should use them as DPDK interfaces.")

Reply:
It was meant to be more like comment generally describing a reason of the function.
Fixed by your style.